nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/PutAzureBlobStorage_v12.java

```diff
@@ -191,4 +198,13 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             session.transfer(flowFile, REL_FAILURE);
         }
     }
+
+    private static void applyUploadResultAttributes(final Map<String, String> attributes, final BlockBlobItem blob, final BlobType blobType, final long length) {
+        attributes.put(ATTR_NAME_BLOBTYPE, blobType.toString());
+        attributes.put(ATTR_NAME_ETAG, blob.getETag());
+        attributes.put(ATTR_NAME_LENGTH, String.valueOf(length));
+        attributes.put(ATTR_NAME_TIMESTAMP, String.valueOf(blob.getLastModified()));
+        attributes.put(ATTR_NAME_LANG, null);
```

Comment:
@malthe According the documentation for [Map.computeIfAbsent()](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#computeIfAbsent-K-java.util.function.Function-), a value of `null` will cause the function to be called. With that being the case, it seems like a value is necessary here to avoid calling `blobClient.getProperties()` in the Supplier for `applyBlobMetadata()`. What do you think?

Reply:
Nice catch, I have made a special case out of it in c5fa8a619a0ab41524daf83eededa7bdcea388fe.